modules/dataquery/ajax/saveQuery.php

```diff
@@ -20,8 +20,29 @@
 $client = new NDB_Client();
 $client->makeCommandLine();
 $client->initialize(__DIR__ . "/../../../project/config.xml");
-$user         = User::singleton();
+
+$user = User::singleton();
+$cdb  = CouchDB::singleton();
+$qid  = $user->getUserName() . "_" . $_REQUEST['QueryName'];
+
+if ($_REQUEST['SharedQuery'] === "true") {
+    $qid = "global:" . $qid;
+}
+
+if ($_REQUEST['OverwriteQuery'] === "false") {
+    $results = $cdb->getDoc(
+        $qid
```

Comment:
I feel like it would not matter if you were to pass the input `validID\nGET /OurDB/sensitiveConfigInfo`. If you look at the follow of the data, `_constructURL` gets called from `_getRelativeURL` which is asking to write to the socket opened in the line above. The socket is open to couchDB anyways so I don't see how one would get sensitive information from on server. If they can the this issue is a lot larger scope then this pull request. 

I've tried to get multiple ways but haven't found a way to get that data. Maybe you can give me an example.

As for documentation I could find any but from playing around with this, I was able to save queries with id like `user_test\\nGET /db/user.com_test`